src/routes/getUpdates.ts

```diff
@@ -0,0 +1,148 @@
+import { asNumber, asObject, asString } from 'cleaners'
+import Router from 'express-promise-router'
+
+import { dataStore } from '../db'
+import {
+  asStoreDirectoryDocument,
+  asStoreFileDocument,
+  asStoreRepoDocument,
+  StoreDirectoryDocument,
+  StoreFileDocument,
+  StoreFileMap,
+  StoreRepoDocument
+} from '../types'
+import { getNameFromPath, makeApiClientError } from '../utils'
+
+export const getUpdatesRouter = Router()
+
+type GetUpdatesBody = ReturnType<typeof asGetUpdatesBody>
+const asGetUpdatesBody = asObject({
+  repoId: asString,
+  timestamp: asNumber
+})
+
+interface GetUpdatesResponseData {
+  paths: StoreFileMap
+  deleted: StoreFileMap
+}
+
+interface DirectoryLike {
+  paths: StoreFileMap
+  deleted: StoreFileMap
+}
+
+getUpdatesRouter.post('/getUpdates', async (req, res) => {
+  let body: GetUpdatesBody
+  let repoId: string
+  let timestamp: number
+
+  try {
+    body = asGetUpdatesBody(req.body)
+    repoId = body.repoId
+    timestamp = body.timestamp
+  } catch (error) {
+    throw makeApiClientError(400, error.message)
+  }
+
+  const repoKey = `${repoId}:/`
+  const repoDocumentResult = await dataStore.get(repoKey)
+
+  const responseData: GetUpdatesResponseData = {
+    paths: {},
+    deleted: {}
+  }
+
+  let repoDocument: StoreRepoDocument
+
+  try {
+    repoDocument = asStoreRepoDocument(repoDocumentResult)
+  } catch (err) {
+    throw makeApiClientError(422, `'${repoKey}' is not a repo`)
+  }
+
```

Comment:
There should be a quick early exit if the repo timestamp is less than or equal to the client timestamp